common/types/ballot.go

```diff
@@ -135,11 +135,11 @@ type Votes struct {
 	// Base ballot.
 	Base BallotID
 	// Support block id at a particular layer and height.
-	Support []Vote
+	Support []Vote `scale:"max=800"` // TODO(mafa): check if this is the right max size
 	// Against previously supported block.
-	Against []Vote
+	Against []Vote `scale:"max=800"` // TODO(mafa): check if this is the right max size
 	// Abstain on layers until they are terminated.
-	Abstain []LayerID
+	Abstain []LayerID `scale:"max=800"` // TODO(mafa): check if this is the right max size
```

Comment:
can @dshulyak @fasmat tell me the rough math of how these max values are derived?

Reply:
i mean 800 sounds excessive for VotingEligibility (considering we are aiming total 50 proposals per layer)
for Support/Against/Abstain this means we will not find a reasonable base ballot for 800 layers?

Reply:
whenever I didn't know what to use as max value I just put a value there that I thought was reasonable and added a `TODO`. I can lower the values to `50` each if that is a better maximum value?